### ISSUE TITLE
multiple instance destroy II

### ISSUE THREAD

**Issue by myblackboxgithub** [liferay/alloy-editor#556]
issue #555 fixed and closed by @ipeyche, thanks, 
but I think it wasn't a good example to control multiple instances.
I have put another code and example.

Once it has crashed like getRanges error, it doesn't work alloyeditor's bridge for ckeditor plugin(like font, fontsize...)
so I would like to fix the issue.
the code is https://1drv.ms/u/s!Am_TIh273u0uhSZ4rF3_F7esnko4

![getranges](https://cloud.githubusercontent.com/assets/15113714/16337497/e6fd53a2-3a0d-11e6-8f09-5b0f84f1bd0e.gif)

Once it has destroy, with the way
```
var editor = AlloyEditor.editable('editable');
editor.destroy();
```
CKEDITOR.instances has Object {}
but editor.get('nativeEditor') exists with name of 'editable'.
Would you look at this?
Thanks.

**Comment by ipeychev** [comment]
I believe the problem is in CKEditor's engine. In more details [here](https://github.com/ckeditor/ckeditor-dev/blob/4.5.9/plugins/clipboard/plugin.js#L1091) - `sel` is null. I think it is null, because triggered from [here](https://github.com/ckeditor/ckeditor-dev/blob/4.5.9/plugins/clipboard/plugin.js#L649) in this `setTimeout` statement.
Would like to open a bug in CKEditor's repository and see what they think?

Thanks,

**Comment by myblackboxgithub** [comment]
Thanks @ipeychev, 
but the error getRanges also happens with text selection, not only with clipboard plugins.

![getrange2](https://cloud.githubusercontent.com/assets/15113714/16356542/a580b56a-3ad3-11e6-9caa-9c60b6312bdc.gif)

I guess when it has destroyed editor and enabled, alloyeditor seems look for the previous range. 
but it has destroyed and get null.

I got the code sample from http://sdk.ckeditor.com/samples/inline.html.
from the sample, the ckeditor inline hasn't any problem for this behavior, 
so I think this error comes from alloyeditor.

**Comment by ipeychev** [comment]
Clipboard plugin subscribes to `mouseup` event, so the type of the selection doesn't matter here.
You did the same with CKEditor only and the error didn't appear?Interesting, could you please upload the code?
If this happens with AlloyEditor only, then I will dig further, but I don't see now how would be that possible. CKEditor does not have any other API than `destroy` and that is what we call. Everything else is in the engine, including this subscription to `mouseup` event.

Thanks,

**Comment by myblackboxgithub** [comment]
thanks for help.
I have uploaded the code of the ckeditor inline.
https://1drv.ms/u/s!Am_TIh273u0uhSh4rF3_F7esnko4

**Comment by ipeychev** [comment]
Thanks, I'm working on this.

**Comment by ipeychev** [comment]
Hey @myblackboxgithub,

Could you please check if the latest version on master fixes the issue for you too?

Thanks,

**Comment by myblackboxgithub** [comment]
Thanks, @ipeychev,
the problem has gone and works as it expected. 
there is a small issue on enable and disable editor with ui-bridge plugins. 
if there is a ui-bridge plugins extra, like Font, FontSize, it keeps complains.
but I will leave it to the next version. 

Warning: flattenChildren(...): Encountered two children with the same key, `.$Font`. Child keys must be unique; when two children share a key, only the first child will be used.
Warning: flattenChildren(...): Encountered two children with the same key, `.$FontSize`. Child keys must be unique; when two children share a key, only the first child will be used.
Uncaught TypeError: Cannot read property 'document' of null

**Comment by ipeychev** [comment]
Okay, will release v1.2.3 then with this fix.

Thanks,

**Comment by ipeychev** [comment]
Fixed in [v1.2.3](https://github.com/liferay/alloy-editor/releases/tag/v1.5.3).
